autoscheduler/frontend/src/utils/throttle.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * Function to throttles the execution of other functions
+ * @param id ID to use, each ID operates on a separate timer
+ * @param callback Callback to run after delay expires
+ * @param delay Delay (in ms) before running effect
+ * @param runOtherIDs Whether to immediately run pending callbacks for IDs other than specified id
```

Comment:
What is the purpose of this parameter?

Reply:
It makes it so only one ID can be active at once, since it runs all of the others immediately if the parameter is true. Right now it's being used so that whenever you switch terms, the courses are saved for other terms immediately rather than having to wait for their intervals to run out (and if you select the old timer before the interval is over, lose your last batch of changes).